pandas/io/formats/style.py

```diff
@@ -422,7 +433,7 @@ def format_attr(pair):
             table_attributes=table_attr,
         )
 
-    def format(self, formatter, subset=None, na_rep: Optional[str] = None):
+    def format(self, formatter, subset=None, na_rep: Optional[str] = None) -> "Styler":
```

Comment:
@simonjayhawkins Have any suggestion on how to annotate ```subset```?

Or in other words how to represent ```IndexSlice``` in a typing form?

Reply:
from pandas.core.indexing import _IndexSlice (maybe in an if TYPE_CHECKING although not necessary)

```suggestion
    def format(self, formatter, subset: Optional[_IndexSlice]=None, na_rep: Optional[str] = None) -> "Styler":
```

Reply:
```mypy``` is raising errors.

I stopped after a few, the error messages got pretty linear.

Any suggestions @simonjayhawkins ?

---

#### MYPY's logs:

For

```python
subset: Optional[_IndexSlice]
```

```
$ mypy pandas
pandas/io/formats/style.py:486: error: Argument 1 to "len" has incompatible type "Optional[_IndexSlice]"; expected "Sized"
pandas/io/formats/style.py:487: error: Incompatible types in assignment (expression has type "Tuple[Optional[_IndexSlice], Any]", variable has type "Optional[_IndexSlice]")
Found 2 errors in 1 file (checked 844 source files)
```

---

For

```python
subset: Tuple[Optional[_IndexSlice], Any]
```

```
$ mypy pandas
pandas/io/formats/style.py:436: error: Incompatible default for argument "subset" (default has type "None", argument has type "Tuple[Optional[_IndexSlice], Any]")
pandas/io/formats/style.py:487: error: Incompatible types in assignment (expression has type "Tuple[Tuple[Optional[_IndexSlice], Any], Any]", variable has type "Tuple[Optional[_IndexSlice], Any]")
Found 2 errors in 1 file (checked 844 source files)
```

---

For 

```python
subset: Tuple[Tuple[Optional[_IndexSlice], Any], Any]
```

```
$ mypy pandas
pandas/io/formats/style.py:436: error: Incompatible default for argument "subset" (default has type "None", argument has type "Tuple[Tuple[Optional[_IndexSlice], Any], Any]")
pandas/io/formats/style.py:487: error: Incompatible types in assignment (expression has type "Tuple[Tuple[Tuple[Optional[_IndexSlice], Any], Any], Any]", variable has type "Tuple[Tuple[Optional[_IndexSlice], Any], Any]")
Found 2 errors in 1 file (checked 844 source files)
```

---

For

```python
subset: Tuple[Tuple[Tuple[Optional[_IndexSlice], Any], Any], Any]
```

```
$ mypy pandas
pandas/io/formats/style.py:436: error: Incompatible default for argument "subset" (default has type "None", argument has type "Tuple[Tuple[Tuple[Optional[_IndexSlice], Any], Any], Any]")
pandas/io/formats/style.py:487: error: Incompatible types in assignment (expression has type "Tuple[Tuple[Tuple[Tuple[Optional[_IndexSlice], Any], Any], Any], Any]", variable has type "Tuple[Tuple[Tuple[Optional[_IndexSlice], Any], Any], Any]")
Found 2 errors in 1 file (checked 844 source files)
```